### PR TITLE
chore: add xray html report generation to nightly pipeline.

### PR DESCRIPTION
**Pull request by blaisedias**

### CODE REVIEW

File: pipelines/Mayastor-e2e-MCP-nightly-stable/Jenkinsfile [hunk 1]
```diff
@@ -59,21 +61,30 @@ pipeline {
   triggers {
     cron(cron_schedule)
   }
+  parameters {
+    choice(choices: ['hc1-nightly','experiment'], name: 'profile', description: 'test profile')
```
Comment (chriswldenyer) [inline pipelines/Mayastor-e2e-MCP-nightly-stable/Jenkinsfile:65]: Is it possible to use the constant e2e_test_profile here?

Comment (chriswldenyer) [reply pipelines/Mayastor-e2e-MCP-nightly-stable/Jenkinsfile:65]: ...or have 'hc1-nightly' as a constant somewhere.

File: pipelines/Mayastor-e2e-MCP-nightly-stable/Jenkinsfile [hunk 2]
```diff
@@ -35,6 +35,8 @@ def e2e_params = [
     xray_send_report : true,
     // Xray: test plan
     xray_test_plan : 'MQ-2184',
+    // Xray: reports directory
+    xray_reports_dir : 'artifacts/xray-report/',
```
Comment (chriswldenyer) [inline pipelines/Mayastor-e2e-MCP-nightly-stable/Jenkinsfile:39]: this looks like it is not used

File: pipelines/Mayastor-e2e-MCP-nightly-stable/Jenkinsfile [hunk 3]
```diff
@@ -237,11 +248,39 @@ pipeline {
...
             }
         }
     }
+    stage('html report'){
+        agent { label 'nixos' }
+        steps {
+            catchError(buildResult: 'SUCCESS', stageResult: 'FAILURE') {
+                script {
+                    def xray_test_plan = e2e_params['xray_test_plan']
+                    def xray_project = xray_test_plan.substring(0,2)
```
Comment (chriswldenyer) [inline pipelines/Mayastor-e2e-MCP-nightly-stable/Jenkinsfile:264]: Assumes the prefix is 2-letters, it doesn't have to be e.g. CAS-...

File: pipelines/Mayastor-e2e-MCP-nightly-stable/Jenkinsfile [hunk 4]
```diff
@@ -237,11 +248,39 @@ pipeline {
...
+                        withCredentials([string(credentialsId: 'XRAY_CLIENT_SECRET', variable: 'XRAY_CLIENT_SECRET')]) {
+                        cmd = "./scripts/xray-report.py --collect ${xray_test_plan} --project ${xray_project}"
+                        sh "nix-shell --run '${cmd}'"
+                        }
+                    }
+                    def files = sh(script: "find artifacts/ -type f", returnStdout: true).split()
+                    if (files.size() != 0) {
+                        def artefacts_stash_queue = e2e_params['artefacts_stash_queue']
+                        def stash_name = "xray-report"
+                        stash includes: 'artifacts/**/**', name: stash_name
```
Comment (chriswldenyer) [inline pipelines/Mayastor-e2e-MCP-nightly-stable/Jenkinsfile:275]: Should this be looking in artifacts/xray-report ?

**Review by chriswldenyer — APPROVED** [review]
LGTM but some comments

File: pipelines/Mayastor-e2e-MCP-nightly-stable/Jenkinsfile (hunk 4 above)
Comment (blaisedias) [reply pipelines/Mayastor-e2e-MCP-nightly-stable/Jenkinsfile:275]: That was the intention, but is unnecessary - it is simpler this way.

**Pull request merged**